packages/runtime/lib/worker/itc.js

```diff
@@ -0,0 +1,154 @@
+'use strict'
+
+const { once } = require('node:events')
+const { parentPort } = require('node:worker_threads')
+
+const { printSchema } = require('graphql')
+const { ITC } = require('@platformatic/itc')
+
+const errors = require('../errors')
+
+function setupITC (app, service, dispatcher) {
+  const itc = new ITC({ port: parentPort })
+
+  itc.handle('start', async () => {
+    const status = app.getStatus()
+
+    if (status === 'starting') {
+      await once(app, 'start')
+    } else {
+      await app.start()
+    }
+
+    if (service.entrypoint) {
+      await app.listen()
+    }
+
+    dispatcher.replaceServer(app.appConfig.useHttp ? app.server.url : app.server)
+
+    return service.entrypoint ? app.server.url : null
+  })
+
+  itc.handle('stop', async () => {
+    const status = app.getStatus()
+
+    if (status === 'starting') {
+      await once(app, 'start')
+    }
+
+    if (status !== 'stopped') {
+      await app.stop()
+    }
+
+    setImmediate(() => {
+      dispatcher.interceptor.close()
+      itc.close()
+    })
+  })
+
+  itc.handle('getStatus', async () => {
+    return app.getStatus()
+  })
+
+  itc.handle('getServiceInfo', async () => {
+    const type = app.config?.configType
+    const version = app.config?.app?.configManagerConfig.version ?? null
+
+    return { type, version }
+  })
+
+  itc.handle('getServiceConfig', async () => {
+    const current = app.config.configManager.current
+    const { logger: _logger, ...server } = current.server
+
+    // Remove all undefined keys from the config
+    return JSON.parse(JSON.stringify({ ...current, server: { ...server, logger: undefined } }))
+  })
+
+  itc.handle('getServiceOpenAPISchema', async () => {
+    if (typeof app.server.swagger !== 'function') {
+      return null
+    }
+
+    try {
+      await app.server.ready()
+      return app.server.swagger()
+    } catch (err) {
+      throw new errors.FailedToRetrieveOpenAPISchemaError(service.id, err.message)
+    }
+  })
+
+  itc.handle('getServiceGraphQLSchema', async () => {
+    if (typeof app.server.graphql !== 'function') {
+      return null
+    }
+
+    try {
+      await app.server.ready()
+      return printSchema(app.server.graphql.schema)
+    } catch (err) {
+      throw new errors.FailedToRetrieveGraphQLSchemaError(service.id, err.message)
+    }
+  })
+
+  itc.handle('getMetrics', async (format) => {
+    const promRegister = app.server.metrics?.client?.register
+
+    if (!promRegister) {
+      return null
+    }
+
+    return format === 'json' ? promRegister.getMetricsAsJSON() : promRegister.metrics()
+  })
+
+  itc.handle('inject', async (injectParams) => {
+    const { statusCode, statusMessage, headers, body } = await app.server.inject(injectParams)
+
+    return { statusCode, statusMessage, headers, body }
+  })
+
+  app.on('changed', () => {
+    itc.notify('changed')
+  })
+
+  return itc
+}
+
+function makeITCSafe (id, worker, itc) {
```

Comment:
I think this should be in itc

Reply:
I followed a different route. Now it doesn't monkey patch anymore.